course7/homeworks/mart/complex/complex.cc

```diff
@@ -0,0 +1,182 @@
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <stack>
+#include <regex>
+
+struct Complex
+{
+	Complex() = default;
+	Complex(double r) : real(r) {}
+	Complex(double r, double i) : real(r) , imaginary(i) {}
+	~Complex() = default;
+	Complex(const Complex& src) : real(src.real) , imaginary(src.imaginary) {}
+
+	Complex& operator=(const Complex& src)
+	{
+		real = src.real;
+		imaginary = src.imaginary;
+		return *this;
+	}
+
+	Complex operator+(const Complex& rhs) const
+	{
+		return { real + rhs.real, imaginary + rhs.imaginary };
+	}
+
+	Complex operator-(const Complex& rhs) const
+	{
+		return { real - rhs.real, imaginary - rhs.imaginary };
+	}
+
+	Complex operator*(const Complex& rhs) const
+	{
+		return {
+			(real * rhs.real) - (imaginary * rhs.imaginary),
+			(real * rhs.imaginary) + (imaginary * rhs.real)
+		  };
+	}
+
+	Complex operator/(const Complex& rhs) const
+	{
+		return {
+			((real * rhs.real) + (imaginary * rhs.imaginary)) / ((rhs.real * rhs.real) + (rhs.imaginary * rhs.imaginary)),
+			((imaginary * rhs.real) - (real * rhs.imaginary)) / ((rhs.real * rhs.real) + (rhs.imaginary * rhs.imaginary))
+		  };
+	}
+
+	double real = 0.0;
+	double imaginary = 0.0;
+};
+
+bool operator==(Complex lhs, Complex rhs)
```

Comment:
Why have you decided to implement this operator as a free function, if other operators are methods? Why the inconsistency?

Reply:
Ah yes, I took it straight from the meeting03 code but forgot to fix this one